core/modules/qhttp/Server.cc

```diff
@@ -104,38 +110,62 @@ void Server::setRequestTimeout(std::chrono::milliseconds const& timeout)
 }
 
 
-void Server::accept()
+void Server::_accept()
 {
-    // Stop accepting new requests if the server has to be stopped
-
-    if (_stopRequested.exchange(false)) {
-        return;
-    }
-
     auto socket = std::make_shared<ip::tcp::socket>(_io_service);
+    {
+        std::lock_guard<std::mutex> lock(_activeSocketsMutex);
+        auto removed = std::remove_if(
+            _activeSockets.begin(),
+            _activeSockets.end(),
+            [](auto& weakSocket) {
+                return weakSocket.expired();
+            }
+        );
+        _activeSockets.erase(removed, _activeSockets.end());
+        _activeSockets.push_back(socket);
+    }
     _acceptor.async_accept(
         *socket,
         [this, socket](boost::system::error_code const& ec) {
-            accept(); // start accept for the next incoming connection
+            _accept(); // start accept for the next incoming connection
             if (!ec) {
-                ip::tcp::no_delay option(true);
-                socket->set_option(option);
+                boost::system::error_code ec;
+                socket->set_option(ip::tcp::no_delay(true), ec);
```

Comment:
The issue is that with the new `stop()` method, client sockets may be closed asynchronously, at any time.  The desired behavior in any of these cases is to have the active handlers tail out, and certainly not to throw.  There is no point recording or reporting errors that result from this path since those error conditions are "on purpose".